app/src/androidTest/java/com/example/concordia_campus_guide/LocationFragmentUITest.java

```diff
@@ -1,27 +1,31 @@
 package com.example.concordia_campus_guide;
 
+import androidx.test.rule.ActivityTestRule;
+import androidx.test.runner.AndroidJUnit4;
+
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import androidx.test.rule.ActivityTestRule;
-import androidx.test.runner.AndroidJUnit4;
-
 import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
-import static androidx.test.espresso.matcher.ViewMatchers.withText;
-import static org.hamcrest.Matchers.allOf;
 
 @RunWith(AndroidJUnit4.class)
-public class UITestExample {
+public class LocationFragmentUITest {
 
     @Rule
     public ActivityTestRule<MainActivity> mActivityRule =
             new ActivityTestRule<>(MainActivity.class);
 
     @Test
-    public void dummyTest(){
-        onView(allOf(withId(R.id.MainActivityText), withText("Hello World!")));
+    public void polygonVisibleTest(){
+        onView(withId(R.id.mapView));
+        onView(withId(R.id.SGWBtn)).perform(click());
+        onView(withId(R.drawable.ev));
+        onView(withId(R.id.loyolaBtn)).perform(click());
+        onView(withId(R.drawable.ad));
```

Comment:
here, I do not think you are verifying much, you should do something like this: onView(withId(...)).check(matches(isDisplayed())). the matcher is important

Reply:
I fixed this and added a new UI test